### PR TITLE
Support Informix db

### PR DESCRIPTION
**Pull request by vivek-22**
Addition of Code to enable the Instana OTEL SDK to support collecting metrics for the Informix DB

### CODE REVIEW

File: rdb/src/main/java/com/instana/dc/rdb/AbstractDbDc.java [hunk 1]
```diff
@@ -91,6 +91,35 @@ public AbstractDbDc(Map<String, Object> properties, String dbSystem, String dbDr
         dbVersion = (String) properties.get(DB_VERSION);
     }
 
+    public AbstractDbDc(Map<String, Object> systemProps, Map<String, String> instanceProps) {
```
Comment (liurui-1) [inline rdb/src/main/java/com/instana/dc/rdb/AbstractDbDc.java:94]: Is this method a required change?

File: rdb/src/main/java/com/instana/dc/rdb/AbstractDbDc.java [hunk 2]
```diff
@@ -236,14 +265,14 @@ public String getDbEntityParentId() {
         return dbEntityParentId;
     }
 
-    public void setDbEntityParentId(String dbEntityParentId) {
```
Comment (liurui-1) [inline rdb/src/main/java/com/instana/dc/rdb/AbstractDbDc.java:239]: This is not a required change.

File: rdb/src/main/java/com/instana/dc/rdb/impl/InformixDc.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+ * (c) Copyright IBM Corp. 2024
+ * (c) Copyright Instana Inc.
+ */
+package com.instana.dc.rdb.impl;
+
+import com.instana.dc.CalculationMode;
+import com.instana.dc.rdb.AbstractDbDc;
+import com.instana.dc.rdb.DbDcUtil;
+import com.instana.dc.rdb.util.Constants;
+import com.instana.dc.rdb.util.InformixUtil;
```
Comment (liurui-1) [inline rdb/src/main/java/com/instana/dc/rdb/impl/InformixDc.java:11]: It is not required to for each implementation to have a source file in "impl" package and another in "util" package. They can use any way. I prefer to have a special package for each implementation. For example: "impl/informix" package for informix, "impl/dameng" package for DamengDB. In each package, the implementation can be very different. Not have to use the same JDBC style.

**Review by liurui-1 — COMMENTED** [review]
Hi @vivek-22 , Thanks for this is very good contribution!
I have some comments especially about the file structure.
I prefer to have the files of one implementation in one package which is a more common style and easier to maintain,

File: rdb/src/main/java/com/instana/dc/rdb/AbstractDbDc.java (hunk 1 above)
Comment (vivek-22) [reply rdb/src/main/java/com/instana/dc/rdb/AbstractDbDc.java]: This method has been removed.

File: rdb/src/main/java/com/instana/dc/rdb/impl/InformixDc.java (hunk 3 above)
Comment (vivek-22) [reply rdb/src/main/java/com/instana/dc/rdb/impl/InformixDc.java]: The package structure has been modified

**Review by liurui-1 — APPROVED** [review]

**Pull request merged**